web_widget_child_selector/static/src/js/widget_child_selector.js

```diff
@@ -0,0 +1,100 @@
+odoo.define('web.web_widget_child_selector', function(require) {
+    "use strict";
+
+    var relational_fields = require('web.relational_fields');
+    var field_registry = require('web.field_registry');
+    var core = require('web.core');
+    var qweb = core.qweb;
+    var FieldMany2One = relational_fields.FieldMany2One;
+
+    var FieldChildSelector = FieldMany2One.extend({
+        template: "FieldChildSelector",
+        events: _.extend({}, FieldMany2One.prototype.events, {
+            'click .o_child_selection_button': '_onChildSelectionClick',
+        }),
+        start: function () {
+            // boolean indicating that the content of the input isn't synchronized
+            // with the current m2o value (for instance, the user is currently
+            // typing something in the input, and hasn't selected a value yet).
+
+            this.$input_dropdown = this.$('.o_input_dropdown');
+            this.$input_value = this.$('.o_input_value')
+            return this._super.apply(this, arguments);
+        },
+        _renderReadonly: function () {
+            var value = _.escape((this.m2o_value || "").trim()).split("\n").join("<br/>");
+            this.$el.html(value);
+            if (!this.nodeOptions.no_open && this.value) {
+                this.$el.attr('href', _.str.sprintf('#id=%s&model=%s', this.value.res_id, this.field.relation));
+                this.$el.addClass('o_form_uri');
+            }
+        },
+        _set_childs: function() {
+            var self = this;
+            this.childs = {};
+            this.parents = {};
+            this.$input_dropdown.empty();
+            this.$input_value.empty();
+            var resources = [];
+            if (this.value.res_id)
+                resources = [this.value.res_id]
+            this._rpc({
+                model: this.field.relation,
+                method: 'get_record_direct_childs_parents',
+                args: [ resources, this.nodeOptions],
```

Comment:
Here you should probably pass the domain as a parameter so python can filter which results returns

Reply:
Comment attended